pkg/main/main.go

```
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strconv"
+	"strings"
+)
+
+const serversEnvVar = "SERVERS"
+const serverCmdEnvVar = "SERVER_CMD"
+
+// Constants indicating the types of errors for Dispatch function invocation
+const (
+	InputError    = "InputError"
+	FunctionError = "FunctionError"
+	SystemError   = "SystemError"
+)
+
+var servers []Server
+var router Router
+
+func handler(w http.ResponseWriter, r *http.Request) {
+	reader := r.Body
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(reader)
+	rawBody := buf.Bytes()
+	var body map[string]interface{}
+	err := json.Unmarshal(rawBody, &body)
+	if err != nil {
+		resp := Response{
+			Context: Context{
+				Error: Error{
+					ErrorType: InputError,
+					Message:   "Invalid Input",
+				},
+			},
+		}
+		out, _ := json.Marshal(resp)
+		fmt.Fprintf(w, string(out))
+		return
+	}
+
+	resp, _ := router.Delegate(body)
+
+	out, _ := json.Marshal(resp)
+
+	fmt.Fprintf(w, string(out))
+}
+
+func main() {
+	numServers, err := strconv.Atoi(os.Getenv(serversEnvVar))
+	if err != nil || numServers < 1 {
```

Not sure about ignoring this error. You can separately check if Getenv returns an empty string. But if there is a value set and is not parsable, we should check it and stop the execution, might be a genuine typo (we don't want user to think the server is running with 4 servers only to later find out he/she typed "r4" instead of "4")